x-pack/plugins/cloud_security_posture/public/pages/findings/latest_findings/use_latest_findings.ts

```diff
@@ -53,16 +53,34 @@ export const showErrorToast = (
   else toasts.addDanger(extractErrorMessage(error, SEARCH_FAILED_TEXT));
 };
 
-export const getFindingsQuery = ({ query, sort }: UseFindingsOptions) => ({
-  index: CSP_LATEST_FINDINGS_DATA_VIEW,
-  body: {
+export const getFindingsQuery = ({ query, sort }: UseFindingsOptions) => {
+  return {
+    index: CSP_LATEST_FINDINGS_DATA_VIEW,
     query,
-    sort: [{ [sort.field]: sort.direction }],
+    sort: getSortField(sort),
     size: MAX_FINDINGS_TO_LOAD,
     aggs: getFindingsCountAggQuery(),
-  },
-  ignore_unavailable: false,
-});
+    ignore_unavailable: false,
+  };
+};
+
+const getSortField = (sort: Sort<CspFinding>) => {
+  if (requiredSortingByPainlessScript.includes(sort.field)) {
+    return {
+      _script: {
+        type: 'string',
+        order: sort.direction,
+        script: {
+          source: `doc["${sort.field}"].value.toLowerCase()`,
+          lang: 'painless',
+        },
+      },
+    };
+  }
+  return { [sort.field]: sort.direction };
+};
+
+const requiredSortingByPainlessScript = ['rule.section', 'resource.name', 'resource.type'];
```

Comment:
Hi @ofiriro3,

Forgive me if I need more context here. Would you mind adding a comment about why these fields require such a sorting instead of using a more native Elasticsearch solution like [normalizer](https://www.elastic.co/guide/en/elasticsearch/reference/current/normalizer.html)?

Reply:
Hi @opauloh.

We had a long discussion about how should we solve the sorting problem.
One of the solutions that I suggested was indeed a normalizer, in the end, it was decided to use a painless script.

Long story short, we didn't want to change the schema/mapping.

You are more than welcome to read the discussion -> 
 - https://github.com/elastic/cloudbeat/issues/104